src/AzureAppConfiguration.ts

```diff
@@ -16,12 +16,40 @@ export type AzureAppConfiguration = {
      * @param thisArg - Optional. Value to use as `this` when executing callback.
      */
     onRefresh(listener: () => any, thisArg?: any): Disposable;
-} & IGettable & AppConfigurationData;
+} & IGettable & IHierarchicalData;
 
-interface AppConfigurationData {
-    data: { [key: string]: any };
+interface IHierarchicalData {
+    /**
+     * Convert the Map-styled data structure to hierarchical object properties.
+     * @param options - The options to control the conversion behavior.
+     */
+    toHierarchicalData(options?: HierarchicalDataConversionOptions): Record<string, any>;

```

Comment:
FAQ: What's `Record`?
`Record<K, V>` is a built-in TypeScript utility type, identical to `{ [key: K]: V }`, for convenience.
See https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type

Reply:
Is it possible to have object in JS where property keys are not `string`? I'm not seeing the value of `Record<string, any>`. Perhaps if it was `Record<string, SomeType>` then yes. But with string and any it seems no better than normal object.

Reply:
object also accepts keys of [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) type, which is a unique primitive value with optional description. That's not the case here.

`Record<string, any>` is identical to `{ [key: string]: any }`, which is what we are using now. And yes, Record is more useful especially for limited values of keys. I'm ok with using the latter if Record introduces confusion.

Reply:
What would it be if it wasn't `Record<string, any>`?

Reply:
I meant to use `{ [key: string]: any }` directly without introducing TypeScript built-in utility type Record, as people might not knowing what a "Record" is. Will that look better?